hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/ReadableFileStoreImpl.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.hedera.node.app.service.file.impl;
+
+import static com.hedera.node.app.service.file.impl.FileServiceImpl.BLOBS_KEY;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.FileID;
+import com.hedera.hapi.node.state.file.File;
+import com.hedera.node.app.service.file.FileMetadata;
+import com.hedera.node.app.service.file.ReadableFileStore;
+import com.hedera.node.app.service.mono.utils.EntityNum;
+import com.hedera.node.app.spi.state.ReadableKVState;
+import com.hedera.node.app.spi.state.ReadableStates;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Provides read-only methods for interacting with the underlying data storage mechanisms for
+ * working with Topics.
+ *
+ * <p>This class is not exported from the module. It is an internal implementation detail.
+ */
+public class ReadableFileStoreImpl extends FileStore implements ReadableFileStore {
+    /** The underlying data storage class that holds the topic data. */
+    private final ReadableKVState<EntityNum, File> fileState;
+
+    /**
+     * Create a new {@link ReadableFileStoreImpl} instance.
+     *
+     * @param states The state to use.
+     */
+    public ReadableFileStoreImpl(@NonNull final ReadableStates states) {
+        requireNonNull(states);
+
+        this.fileState = states.get(BLOBS_KEY);
+    }
+
+    /**
+     * Returns the file metadata needed. If the file doesn't exist returns failureReason. If the
+     * file exists , the failure reason will be null.
+     *
+     * @param id file id being looked up
+     * @return file's metadata
+     */
+    // TODO : Change to return File instead of TopicMetadata
```

Comment:
It looks like this TODO has been implemented and can be deleted. Did you go through the SonarLint warnings? This TODO should have been listed as a warning.

Reply:
removed it thanks